Hexastore.Web/Queue/QueueContainer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Hexastore.Web.EventHubs;
+using Microsoft.Extensions.Logging;
+
+namespace Hexastore.Web.Queue
+{
+    public class QueueContainer : IQueueContainer, IDisposable
+    {
+        private readonly EventReceiver _eventReceiver;
+        private readonly QueueWriter[] _queueWriters;
+        private readonly ILogger<QueueContainer> _logger;
+        private readonly int _count = 32;
+        private bool _running = true;
+
+        public QueueContainer(EventReceiver eventReceiver, ILogger<QueueContainer> logger)
+        {
+            _logger = logger;
+            _eventReceiver = eventReceiver;
+            _queueWriters = new QueueWriter[_count];
+            for(int i=0; i< _count; i++) {
+                _queueWriters[i] = new QueueWriter(eventReceiver, logger);
+            }
+            _ = _eventReceiver.LogCount();
+            _ = LogQueueLength();
+        }
+
+        public void Dispose()
+        {
+            _running = false;
+            foreach(var q in _queueWriters) {
+                q.Dispose();
+            }
+        }
+
+        public void Send(StoreEvent storeEvent)
+        {
+            var partitionId = Math.Abs(storeEvent.PartitionId.GetHashCode() % _count);
```

Comment:
GetHashCode() isn't guaranteed to be consistent across implementations/versions. We may way to implement our own to not tie us to a specific version.

Reply:
We should write one. It might have to look at the byte level, previous hash code generators I've used all seems to rely on the underlying string hash to start with: https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/build/Shared/HashCodeCombiner.cs

Reply:
Agreed. I know FnV or JumpHash does a good job. Will take a look at introducing a hashing function.